### PR TITLE
Feature/implement otel feature flag

### PR DESCRIPTION
**Pull request by franmoore05**
### What

Added a feature flag for the otel functionality.

### How to review

Ensure the changes make sense and the tests pass.

### Who can review

Anyone ONS.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Given json defining a table, capable of rendering a table in multiple formats
 | OTEL_EXPORTER_OTLP_ENDPOINT      | localhost:4317                            | Host and port for the OpenTelemetry endpoint                                             |
 | OTEL_SERVICE_NAME                | dp-table-renderer                         | Service name to report to telemetry tools                                                |
 | OTEL_BATCH_TIMEOUT               | 5s                                        | Interval between pushes to OT Collector                                                  |
+| OTEL_ENABLED                 | false                   | Feature flag to enable OpenTelemetry
```
Comment (lindenmckenzie) [inline README.md:19]: formatting :)

File: api/api.go [hunk 2]
```diff
@@ -52,15 +65,28 @@ func createCORSHandler(allowedOrigins string, router *mux.Router) http.Handler {
...
+			// Configure the "http.route" for the HTTP instrumentation.
+			handler := otelhttp.WithRouteTag(pattern, http.HandlerFunc(handlerFunc))
+			api.router.Handle(pattern, handler)
+		}
+
+		handleFunc("/render/{render_type}", api.renderTable)
+		handleFunc("/parse/html", api.parseHTML)
+	} else {
+		api.router.HandleFunc("/render/{render_type}", api.renderTable).Methods("POST")
+		api.router.HandleFunc("/parse/html", api.parseHTML).Methods("POST")
```
Comment (lindenmckenzie) [inline api/api.go:85]: This seems like an error prone way of handling the routing, is there another approach which means we don't have two separate route definitions?

File: main.go [hunk 3]
```diff
@@ -87,7 +91,6 @@ func run(ctx context.Context) error {
 			return err
 		}
 
-		otelShutdown(ctx)
 		log.Info(ctx, "Shutdown complete")
```
Comment (lindenmckenzie) [inline main.go:94]: I don't understand this change.

**Review by lindenmckenzie — COMMENTED** [review]
Very minor comments. 

As a general comment across the piece, the same is true about the config passing / not passing etc. I'm happy to pass it as is, so will leave to you if you want to action that or not.

File: README.md (hunk 1 above)
Comment (franmoore05) [reply README.md]: This should now be updated

File: api/api.go (hunk 2 above)
Comment (franmoore05) [reply api/api.go]: This has been refactored, please let me know if it's better than the previous iteration :)

File: main.go (hunk 3 above)
Comment (franmoore05) [reply main.go:94]: This got moved up to line 65 - it's called when the service shuts down and is otel specific. This ensures the service closes down nicely.

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**